llvm/cmake/modules/GetHostTriple.cmake

```diff
@@ -42,7 +42,21 @@ function( get_host_triple var )
       set( value "powerpc-ibm-aix" )
     endif()
   else( MSVC )
-    if(CMAKE_HOST_SYSTEM_NAME STREQUAL Windows AND NOT MSYS)
+    if(CMAKE_HOST_SYSTEM_NAME STREQUAL Windows AND NOT MSYS AND CMAKE_C_COMPILER_ID MATCHES "Clang" )
```

Comment:
Why is this needed here? When operating in MSVC mode, CMake usually sets the `MSVC` CMake variable, so this should hit the existing codepath at lines 5-18 above? And this is a bit weird, because CMake usually doesn't set `COMPILER_ARCHITECTURE_ID` for Clang (or GCC), but only for MSVC. I guess CMake might set `COMPILER_ARCHITECTURE_ID` for Clang-cl when operating in MSVC mode, but if we're invoking it as `clang`, not `clang-cl` so that the `MSVC` variable isn't set, it feels odd that CMake still would be setting `COMPILER_ARCHITECTURE_ID`.

Also I don't quite see why this needs to be as a separate step within the `else()` clause here, why can't this be a toplevel case like the other ones?

The condition, `CMAKE_HOST_SYSTEM_NAME STREQUAL Windows AND NOT MSYS` also feels quite contrieved for this case. I guess what you want to detect is "clang, targeting windows, probably in msvc mode (as we're past a  check for `MINGW` further up)", and we generally don't want to check for the host unless we need to.

The existing check for `CMAKE_HOST_SYSTEM_NAME STREQUAL Windows AND NOT MSYS` is to pick out configurations where we _can't_  run the generic detection shell script below.


Can't we just extend the existing MSVC case at the top, e.g. something like `if (MSVC OR (CMAKE_C_COMPILER_ID MATCHES "Clang" AND WIN32 AND CMAKE_C_COMPILER_ARCHITECTURE_ID)` or something along those lines, to specifically match the case we're looking for here?